doc/nrf/glossary.rst

```diff
@@ -0,0 +1,447 @@
+.. _glossary:
+
+Glossary
+########
+
+.. contents::
+   :local:
+   :depth: 2
+
+Anycast addressing
+   An addressing type that routes datagrams to a single member of a group of potential receivers that are all identified by the same destination address.
+   This is a one-to-nearest association.
+
+Access Port Protection (APPROTECT)
+   A register used to prevent read and write access to all CPU registers and memory-mapped addresses.
+
+API Result
+   Indicates success or failure of an API procedure.
+   In case of failure, a comprehensive error code indicating the cause or reason for the failure is provided.
+
+Application
+   An implementation example that includes a variety of libraries to implement a specific use case.
+
+Application Programming Interface (API)
+   A language and message format used by an application program to communicate with an operating system, application, or other services.
+
+Board
+   In Zephyr and the |NCS|, a data structure for describing hardware on the board and its configuration files.
```

Comment:
We use different boards for non-secure and secure variants of a build, should we mention that here? Currently it seems to map to hardware only.

Reply:
fixed